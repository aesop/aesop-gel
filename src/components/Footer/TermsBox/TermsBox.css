@import 'styles/index';

.base {
  display: inline-block;
  width: 100%;
  box-sizing: border-box;
  padding-top: 16px;
  padding-right: 7px;
  padding-bottom: 18px;
  padding-left: 14px;
  border: 1px solid var(--color-white);
  margin-top: 23px;
  border-radius: 3px;
}

.term {
  width: 100%;
  height: 90px;
  border: none;
  background: transparent;
  color: var(--color-white);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(12px);
  line-height: 1.67;
  resize: none;
  &::-webkit-scrollbar {
    width: 5px;
    margin-right: -10px;
    -webkit-appearance: none;
  }
  &::-webkit-scrollbar-thumb {
    background-color: var(--color-white);
    border-radius: 4px;
    -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
  }
}
