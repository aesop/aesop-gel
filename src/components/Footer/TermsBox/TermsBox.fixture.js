export default {
  term:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta.',
};
