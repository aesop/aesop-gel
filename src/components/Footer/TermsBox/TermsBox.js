import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './TermsBox.css';

const TermsBox = ({ className, name, term }) => {
  const baseClassSet = cx(styles.base, className);
  const termBoxClassSet = cx(styles.term);
  return (
    <label htmlFor={name}>
      <div className={baseClassSet}>
        <textarea className={termBoxClassSet} disabled={true} value={term} />
      </div>
    </label>
  );
};

TermsBox.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  term: PropTypes.string,
};

TermsBox.defaultProps = {
  className: undefined,
  name: undefined,
  term: undefined,
};

export default TermsBox;
