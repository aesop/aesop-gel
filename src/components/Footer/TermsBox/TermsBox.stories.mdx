import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import Podium from '~/components/Podium';
import TermsBox from './TermsBox';
import TermsBoxFixture from './TermsBox.fixture'

# TermsBox

<Meta title="Components/TermsBox" component={TermsBox} />

<Preview>
  <Story name="Base component">
    <Podium horizontalPadding="small" verticalPadding="small">
      <p>Set the background to dark</p>
      <TermsBox
        name='test-textarea'
        term={TermsBoxFixture.term}
      />
    </Podium>
  </Story>
</Preview>

## API

<Props of={ContentHubArticle} />

