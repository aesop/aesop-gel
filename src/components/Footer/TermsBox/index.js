import TermsBox from './TermsBox';

export default TermsBox;
