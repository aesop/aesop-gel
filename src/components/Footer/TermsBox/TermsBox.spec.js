import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import TermsBox from './TermsBox';
import TermsBoxFixture from './TermsBox.fixture';

configure({ adapter: new Adapter() });

describe('<TermsBox />', () => {
  it('should be defined', () => {
    expect(TermsBox).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(<TermsBox id="test-textarea" term={TermsBoxFixture.term} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
