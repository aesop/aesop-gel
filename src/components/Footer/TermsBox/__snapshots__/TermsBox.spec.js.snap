// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TermsBox /> renders base component correctly 1`] = `
<label
  htmlFor="test-textarea"
>
  <div
    className="base"
  >
    <textarea
      className="term"
      disabled={true}
      id="test-textarea"
      value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta."
    />
  </div>
</label>
`;
